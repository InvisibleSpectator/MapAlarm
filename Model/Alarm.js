export default class Alarm {
  constructor() {
    this.id = 0;
    this.name = 'New Alarm';
    this.time = new Date();
    this.options = [
      {
        name: 'Monday',
        value: false,
      },
      {
        name: 'Tuesday',
        value: false,
      },
      {
        name: 'Wednesday',
        value: false,
      },
      {
        name: 'Thursday',
        value: false,
      },
      {
        name: 'Friday',
        value: false,
      },
      {
        name: 'Saturday',
        value: false,
      },
      {
        name: 'Sunday',
        value: false,
      },
    ];
    this.isActive = true;
    this.isLocationBound = false;
    this.location = {
      latitude: 0,
      longitude: 0,
    };
    this.description = '';
  }
}
