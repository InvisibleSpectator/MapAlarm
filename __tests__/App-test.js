/**
 * @format
 */

import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import DrawerAlarmCArd from '../Pages/Components/DrawerAlarmCard';
import Alarm from '../Model/Alarm';

it('should render component', () => {
  const wrapper = renderer.create(
    <DrawerAlarmCArd alarm={{...new Alarm(), time: new Date(0)}} />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('model test', () => {
  const alarm = new Alarm();
  expect(alarm.options).toStrictEqual([
    {
      name: 'Monday',
      value: true,
    },
    {
      name: 'Tuesday',
      value: true,
    },
    {
      name: 'Wednesday',
      value: true,
    },
    {
      name: 'Thursday',
      value: true,
    },
    {
      name: 'Friday',
      value: true,
    },
    {
      name: 'Saturday',
      value: true,
    },
    {
      name: 'Sunday',
      value: true,
    },
  ]);
});
