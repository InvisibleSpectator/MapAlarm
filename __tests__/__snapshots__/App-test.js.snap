// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    header={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomWidth": null,
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      uppercase={false}
    >
      New Alarm
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      04:00
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 28,
              "width": 35,
            },
            Object {
              "color": "#32505C",
              "position": "absolute",
              "right": 12,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      type="MaterialIcons"
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#007aff",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        type="MaterialIcons"
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#007aff",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        type="MaterialIcons"
      >
        
      </Text>
    </View>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Array [
            Object {
              "marginBottom": -5,
              "marginTop": -5,
            },
            Object {
              "position": "absolute",
              "right": 0,
            },
          ],
        ]
      }
      value={true}
    />
  </View>
</View>
`;
